### PR TITLE
Update website_webhook action to the use of discussions

### PR DESCRIPTION
**Pull request by pattacini**
This PR updates the action in order to be triggered by discussions.
Indeed, GitHub has made available the [discussions events](https://github.blog/changelog/2021-06-29-github-discussions-events-available-on-github-actions).

To this end, I had to apply some changes.

A couple of notes:
- The part on fetching the description is empty now as it's unclear to me what was done before and how we can adapt it to the [new fields](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#discussion).
- It'd be also convenient (maybe better off with a second PR) to use the latest app-oriented way to dispatch info to `icub-tech-iit/code`.

cc @AlexAntn @vtikha @vvasco @lauracavaliere @ilaria-carlini

### CODE REVIEW

**Comment by pattacini** [comment]
Hold on, I'd need to refactor this PR since we don't have [discussion templates][1] yet; thus, it's not possible to clearly identify a summary and a description part.

Therefore, I'm gonna provide as a summary the first let's say 500 characters (truncated) of the discussion body.
The description will be the whole discussion body instead.

Don't know how this will impact the Bazaar backend but it's a first working implementation anyway.

[1]: https://github.com/github/feedback/discussions/2838?sort=top#discussioncomment-261382

**Comment by pattacini** [comment]
Here we are with a new version that handles the truncated body as a possible summary.

That said, I'm frankly skeptical that how the action is conceived is the correct way to go. I can see the following points:
- The action should only dispatch the discussion id. All the processing and data retrieval should be deferred to who actually needs this information.
- Using `bash` to process strings that may contain special chars like `"` and `'` is intrinsically ill-posed. I experienced problems with those cases several times in the past so I decided to switch to Ruby for example as done in [`create-mentioning-comment.yml`](https://github.com/icub-tech-iit/outside-collaborators/blob/master/templates/.github/workflows/create-mentioning-comment.yml), which performs really great in this respect.

To sum up, my impression is that the action should be really small and responsible for dispatching just the discussion id, whereas the recipient retrieves the discussion payload and makes use of Ruby to parse the discussion body in a way that is robust against special chars.

What do you think @AlexAntn ?

**Comment by AlexAntn** [comment]
Ciao @pattacini !

> * The action should only dispatch the discussion id. All the processing and data retrieval should be deferred to who actually needs this information (even the date).
> * Using `bash` to process strings that may contain special chars like `"` and `'` (the combination of the two turns to be very nasty) is intrinsically ill-posed. I experienced problems with those cases several times in the past so I decided to switch to Ruby for example as done in [`create-mentioning-comment.yml`](https://github.com/icub-tech-iit/outside-collaborators/blob/master/templates/.github/workflows/create-mentioning-comment.yml), which performs really great in this respect.

This is a possibility, but it would require quite a bit of refactoring on the website part. I think we can for now integrate this action with your modifications (and with some modifications we tested, I'll include them shortly), and in the meantime we change the website to work with only the ID of the announcements.

**Comment by pattacini** [comment]
Sounds like a good plan.
Let me put the PR ready for review then!

File: .github/workflows/website_webhook.yml [hunk 1]
```diff
@@ -1,67 +1,42 @@
...
-      echo $(cat temp_title.txt)
-      sed -nr 's/.*"title": (.*), "updated.*/\1/p' <temp_title.txt >title.txt
-      echo $(cat title.txt)
-      if [[ $(cat title.txt) == *"Release"* ]]
+      category="${{ github.event.discussion.category.name }}"
+      title="${{ github.event.discussion.title }}"
+      body="${{ github.event.discussion.body }}"
+      
+      echo "::set-output name=title::${title}"
+      echo "::set-output name=summary::${body:0:500}"
```
Comment (AlexAntn) [inline .github/workflows/website_webhook.yml:20]: We can pass only the body, and do the parsing for the summary on the website side (easy change there, and we can remove one output)
```suggestion
```

Comment (pattacini) [reply .github/workflows/website_webhook.yml:20]: The proposed commit would overwrite too many lines! Can you amend it?
Also, we need to update the dispatching part.

File: .github/workflows/website_webhook.yml [hunk 2]
```diff
@@ -1,67 +1,42 @@
...
-  - name: Get description
-    id: description
-    env:
-      ISSUE_CONTEXT: ${{ toJson(github.event.issue) }}
-    run: |
-      echo $ISSUE_CONTEXT > temp.txt
-      sed -nr 's/.*"body": (.*), "closed_at.*/\1/p' <temp.txt >temp5.txt
-      echo "::set-output name=description::$(cat temp5.txt)"  
+        echo "::set-output name=is_release::false"
+      fi
```
Comment (AlexAntn) [inline .github/workflows/website_webhook.yml:28]: We can remove the "release" from the outputs, and instead pass the category directly - we can then process it on the website side. We want to be able to process not only the releases, but also other categories e.g.: summer schools, announcements, etc.

File: .github/workflows/website_webhook.yml [hunk 3]
```diff
@@ -1,67 +1,42 @@
...
 
   - name: Repository Dispatch
-    uses: peter-evans/repository-dispatch@v1
+    uses: peter-evans/repository-dispatch@v1.1.3
     with:
       token: ${{ secrets.CODE_REPO_ACCESS_TOKEN }}
       repository: icub-tech-iit/code
       event-type: announcement_launched
-      client-payload: '{"description": ${{ steps.description.outputs.description }}, "id": "${{ steps.date.outputs.date }}", "title": ${{ steps.title.outputs.title }}, "date": "${{ steps.date.outputs.date_spelled }}", "summary": "${{ steps.summary.outputs.summary }}", "image": "???", "type" : "announcement_launched", "release": "${{ steps.rel_annou.outputs.release }}"}'
+      client-payload: '{"description": ${{ steps.announcement.outputs.description }}, "id": "${{ steps.date.outputs.date }}", "title": ${{ steps.announcement.outputs.title }}, "date": "${{ steps.date.outputs.date_spelled }}", "summary": "${{ steps.announcement.outputs.summary }}", "image": "???", "type" : "announcement_launched", "release": "${{ steps.announcement.outputs.is_release }}"}'
```
Comment (AlexAntn) [inline .github/workflows/website_webhook.yml:42]:
```suggestion
      client-payload: '{"description": "${{ steps.announcement.outputs.description }}", "id": "${{ steps.date.outputs.date }}", "title": "${{ steps.announcement.outputs.title }}", "date": "${{ steps.date.outputs.date_spelled }}", "image": "???", "type" : "announcement_launched", "category": "${{ steps.announcement.outputs.category }}"}'
```

Some changes here:
- We ran some tests and with the changes you did it requires the use of quotes on both the description and title (for the json format);
- We can remove the summary from the payload, since we would handle this from the website side;
- finally, we would replace the "release" payload with "category" instead.

**Comment by pattacini** [comment]
@AlexAntn take a fresh look now.

**Review by AlexAntn — APPROVED** [review]

**Comment by AlexAntn** [comment]
Good for me!
